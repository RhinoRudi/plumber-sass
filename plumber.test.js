describe('plumber-sass', () => {
	it('should render scss', async () => {
		const data = `
		@import 'plumber';
		p {
			@include plumber(
				$grid-height: 1rem,
				$baseline: 0.158203,
				$font-size: 1.75,
				$line-height: 3,
				$margin-top: 1,
				$margin-bottom: 2
			);
		}`;

		const result = await render(data);
		expect(result).toMatchSnapshot();
	});
	it('should throw an error', async () => {
		const data = `
		@import 'plumber';
		p {
			@include plumber(
				$grid-height: 1rem,
				$baseline: 0.158203,
				$font-size: 1.75,
				$line-height: -3,
				$margin-top: 1,
				$margin-bottom: 2
			);
		}`;

		const result = await render(data);
		expect(result).toMatchSnapshot();
	});
});