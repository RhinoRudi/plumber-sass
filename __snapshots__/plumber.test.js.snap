// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plumber-sass should render scss 1`] = `
"p {
  font-size: 1.75rem;
  line-height: 3rem;
  margin-top: 0rem;
  padding-top: 0.901855rem;
  padding-bottom: 0.098145rem;
  margin-bottom: 2rem;
}
"
`;

exports[`plumber-sass should throw an error 1`] = `"$line-height parameter must be a positive integer, got -3 instead"`;
